pybaseball/playerid_lookup.py

```diff
@@ -47,77 +50,116 @@ def chadwick_register(save: bool = False) -> pd.DataFrame:
     table['name_first'] = table['name_first'].str.lower()
     return table
 
 
-def playerid_lookup(last=None, first=None, player_list=None):
-    # force input strings to lowercase
-    if last:
-        last = last.lower()
-    if first:
-        first = first.lower()
-    table = get_lookup_table()
-
-    # if player_list has a value, then the user is passing in a list of players
-    # the list of players may be comma delimited for last, first, or just last
-    if player_list:
-        player_counter = 1
-        for player in player_list:
-            last = player.split(",")[0].strip()
-            first = None
-            if (len(player.split(",")) > 1):
-                first = player.split(",")[1].strip()
-            if (player_counter == 1):
-                results = playerid_lookup(last, first)
-            else:
-                results = results.append(playerid_lookup(last, first), ignore_index=True)
-            player_counter += 1
+def name_similarity(last: str, first: str, player_table: pd.DataFrame) -> pd.DataFrame:
+    """Calculates similarity of first and last name provided with all players in player_table
+
+    Args:
+        last (str): Provided last name
+        first (str): Provided first name
+        player_table (pd.DataFrame): Chadwick player table including names
+
+    Returns:
+        pd.DataFrame: 5 nearest matches from fuzzywuzzy.process
+    """
+    filled_df = player_table.fillna("")
+    chadwick_names = filled_df["name_first"] + " " + filled_df["name_last"]
+    fuzzy_matches = pd.DataFrame(process.extract(f"{first} {last}", chadwick_names, limit=5))
+    matched_names = fuzzy_matches[0].str.split(expand=True)
+    matched_names = matched_names.rename(columns={0: "name_first", 1: "name_last"})
+    return matched_names
+
+
+class player_search_client:
+    def __init__(self) -> None:
+        self.table = get_lookup_table()
+
+    def search(self, last: str = None, first: str = None, player_list: List[str] = None, fuzzy: bool = False) -> pd.DataFrame:
+        """Lookup playerIDs (MLB AM, bbref, retrosheet, FG) for a given player
+
+        Args:
+            last (str, optional): Player's last name. Defaults to None.
+            first (str, optional): Player's first name. Defaults to None.
+            player_list (list, optional): List of players to seach for (comma delimited). Defaults to None.
+            search (bool, optional): In case of typos, returns players with names close to input. Defaults to False.
+
+        Returns:
+            pd.DataFrame: DataFrame of playerIDs, name, years played
+        """
+
+        # force input strings to lowercase
+        if last:
+            last = last.lower()
+        if first:
+            first = first.lower()
+
+        # if player_list has a value, then the user is passing in a list of players
+        # the list of players may be comma delimited for last, first, or just last
+        if player_list:
```

Comment:
do we need to support a list? I'd suggest just getting rid of this and restore it if /when someone asks for it. alternatively make processing a list an explicitly different method? I basically think the difference in the way the arguments are inputted is confusing, like if you want one player you enter `last=last, first=first`, but if you want a list you enter comma separated strings so I'd prefer they're consistent or a list is a different method

Reply:
True. We could unwind this to a separate method -- perhaps just tuples of the pairs? And we can definitely unwind this.